app/mqtt_reader.py

```diff
@@ -61,37 +56,35 @@ def run(self) -> None:
         self._client.on_connect = self._on_connect()
         self._client.on_message = self._on_message()
 
-        try:
-            self._client.connect(self.host, self.port)
-        except OSError as e:
-            raise OSError(
-                f"Cannot assign requested address: {self.host}:{self.port}"
-            ) from e
+        self.connect()
 
         logging.info("Service started")
         self._client.loop_forever()
 
-    def _on_connect(self):
-        def inner(client, _userdata, _flags, rc):
-            if rc == 0:
-                logging.info("Connected to MQTT broker")
-                for topic in self.topics:
-                    logging.info("Subscribing to topic: %s", topic)
-                    client.subscribe(topic)
-            else:
-                logging.error("Connection failed")
-
-        return inner
-
     def _on_message(self):
         def inner(_client, _userdata, message):
             try:
                 msg_str = _decode_message(message)
                 msg_obj = CommandMessage.read(msg_str)
             except InvalidMessageError as ex:
                 logging.error(ex)
+                self.error_handler.publish(
+                    self.error_handler.Category.INVALID_MESSAGE, str(ex)
+                )
                 return
             for callback in self.on_message_callbacks:
                 callback(msg_obj)
 
         return inner
+
+    def _on_connect(self):
+        def inner(client, _userdata, _flags, rc):
+            if rc == 0:
+                logging.info("Connected to MQTT broker")
+                for topic in self.topics:
+                    logging.info("Subscribing to topic: %s", topic)
+                    client.subscribe(topic)
+            else:
+                logging.error("Connection failed")
```

Comment:
Shouldn't we be throwing an exception here ?

Reply:
This is just the callback after connect is completed - if it had failed we would already have raised an `OSError` exception in the main `connect` method